### PR TITLE
add generic details command

### PR DESCRIPTION
**Pull request by ryanfkeepers**
centralizes details command processing in the cli

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #2025

### CODE REVIEW

File: src/cli/backup/sharepoint_test.go [hunk 1]
```diff
@@ -391,51 +387,3 @@ func (suite *SharePointUnitSuite) TestSharePointBackupCreateSelectors() {
 		})
 	}
 }
-
-func (suite *SharePointUnitSuite) TestSharePointBackupDetailsSelectors() {
```
Comment (vkamra) [inline src/cli/backup/sharepoint_test.go:395]: Do we have coverage for sharepoint details now? The change in `backup_test` only includes exchange. Can we leave this test in here?

**Comment by vkamra** [comment]
Change looks good. Question on test coverage for sharepoint/onedrive/groups.

**Review by vkamra — APPROVED** [review]
Change LGTM. Question on test coverage and whether we need to leave the existing tests or add a few more test cases in `backup_test.go`

**Review by pandeyabs — APPROVED** [review]

File: src/cli/backup/sharepoint_test.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/cli/backup/sharepoint_test.go:395]: This wasn't exercising integration behavior in details, it was just a roundabout way of testing the selectors behavior via mocks, all of which is already covered in other sections. The `backup_test` reference to exchange doesn't exercise exchange in any particular way either.  We primarily need a valid service to stand in place there.

As separate toil, I'd be happy to go through and re-evaluate test coverage after these cases for security, however.  They seem to have grow in a couple different directions across time and could use some re-alignment.

**Pull request merged**